src/helpers/updateSearch.tsx

```diff
@@ -0,0 +1,47 @@
+export type SearchParams = {
+  [key: string]: string | string[] | null,
+};
+
+/**
+ * This function prepares a correct search string
+ * from a given currentParams and paramsToUpdate.
+ */
+export function updateSearch(
+  currentParams: URLSearchParams,
+  paramsToUpdate: SearchParams, // it's our custom type
+): string {
+  // copy currentParams by creating new object from a string
+  const newParams = new URLSearchParams(
+    currentParams.toString(),
+  );
+
+  // Here is the example of paramsToUpdate
+  // {
+  //   sex: 'm',                ['sex', 'm']
+  //   order: null,             ['order', null]
+  //   centuries: ['16', '19'], ['centuries', ['16', '19']]
+  // }
```

Comment:
You can leave these ones. They look useful)

Reply:
Thanks for check, all done